VideoWeb/VideoWeb/ClientApp/src/app/security/authentication/quick-link-security.service.spec.ts

```diff
@@ -0,0 +1,217 @@
+import { fakeAsync, flush } from '@angular/core/testing';
+import { Subject } from 'rxjs';
+import { ApiClient } from 'src/app/services/clients/api-client';
+import { JwtHelperService } from '../jwt-helper.service';
+import { QuickLinkJwtBody, QuickLinkSecurityService } from './quick-link-security.service';
+
+describe('QuickLinkSecurityService', () => {
+    let service: QuickLinkSecurityService;
+    let apiClientSpy: jasmine.SpyObj<ApiClient>;
+    let jwtHelperSpy: jasmine.SpyObj<JwtHelperService>;
+    const jwt =
+        'eyJhbGciOiJIUzUxMiIsInR5cCI6IkpXVCJ9.eyJ1bmlxdWVfbmFtZSI6IkpvaG4gRG9lIiwiZ2l2ZW5fbmFtZSI6IkpvaG4gRG9lIiwiZmFtaWx5X25hbWUiOiJKb2huIERvZSIsInByZWZlcnJlZF91c2VybmFtZSI6IkpvaG4gRG9lIiwicm9sZSI6IkNpdGl6ZW4iLCJuYmYiOjE2MjcyOTQwMzcsImV4cCI6MTYyNzMyMjk1NywiaWF0IjoxNjI3Mjk0MDk3LCJpc3MiOiJodHRwczovL3ZoLXZpZGVvLXdlYi1kZXYuYXp1cmV3ZWJzaXRlcy5uZXQvOThhNWRiM2QtMGY5MS00MDNmLWI3ZGMtZDFhMjcyZjQ2ZjNiIn0.NyH-9u3Vg2wSC-B2rxkqjbAbKvdvoCyyFAgBsfeP9ff9mQTxn6PfJHdtkp8sANnQHpsLdqW8VnAp9a9bTfTVDA';
+    const decodedJwt = {
+        unique_name: 'John Doe',
+        given_name: 'John Doe',
+        family_name: 'John Doe',
+        preferred_username: 'John Doe',
```

Comment:
Make these different to (hopefully) catch any issues where the wrong value is mapped.

Reply:
The token uses mainly preferred_username